modules/vpc/locals.tf

```diff
@@ -1,3 +1,33 @@
 locals {
   vpc_name = "${var.namespace}-vpc"
+
+  azs = [
+    "ap-southeast-1a",
+    "ap-southeast-1b",
+  ]
+
+  private_subnets = [
+    "10.0.1.0/24",
+    "10.0.2.0/24",
+  ]
+  public_subnets = [
+    "10.0.3.0/24",
+    "10.0.4.0/24",
+  ]
+
+  # Single NAT Gateway scenario
+  # all private subnets will route their Internet traffic through this single NAT gateway.
+  # The NAT gateway will be placed in the first public subnet in the public_subnets block.
+
+  # provision NAT Gateways for each of private networks
+  enable_nat_gateway = true
```

Comment:
I think we need a single NAT gateway, and all requests from private subnets will go through it. That's what the diagram illustrated, right?

Reply:
There's no NAT Gateway in your diagram.

> I think we need a single NAT gateway, and all requests from private subnets will go through it

Only if the application needs to initiate a connection to the external network, ie updates/patches, external API etc. For the IC with ECS Fargate accessing RDS and Elasticache I don't think it's needed. S3 will also be accessed on client side.

Maybe we can keep it for now as in the real-world it would probably be common to have one indeed. It's not mentioned in the IC requirements so maybe this needs to be added.

Reply:
thanks for pointing out Liam 🙏, I think I should keep it since we are all using NAT Gateway for routing 👍

Reply:
It may need to initiate a connection the external network after all if in prviate - ECS might need to do this to connect to ECR to get the image 😆

Reply:
It seems for case mentioned above - accessing the ECR image requireing outbound connection, is normally best handled with VPC endpoint and not NAT Gateway:

- https://www.reddit.com/r/aws/comments/ul4b0e/cant_pull_an_image_from_ecr_in_a_private_subnet/

Also, Xavier made a presentation mentioning this:
https://docs.google.com/presentation/d/1PZMW8tFrzrnGEo_Zt9FprRlNYx-ZzLX0svpeJtkoZJg/edit#slide=id.g280aee4c295_0_105